### ISSUE TITLE
variable not supported in server url within openapi 3.1.0

### ISSUE THREAD

**Issue by fidgi** [stoplightio/spectral#1911]
**Describe the bug**
When a variable is used in a server url within an openapi 3.1.0 document as follow:
```
openapi: 3.1.0
...
servers:
- url: https://api.example.com/{version}
  description: |
    API location
  variables:
    version:
      default: v1

```
, the following error is reported by spectral :

```
spectral lint specs/openapi.yaml


/home/me/udd/public-profiles-api/specs/openapi.yaml
 103:8  error  oas3-schema  "url" property must match format "uri".  servers[0].url

✖ 1 problem (1 error, 0 warnings, 0 infos, 0 hints)
```

All is fine if the variable is removed or if the openapi version is set to 3.0.3 .

**To Reproduce**
see above

**Expected behavior**

Spectral should not report an error in this case

**Environment (remove any that are not applicable):**
 - Library version: @stoplight/spectral-cli 6.1.0
 - OS: Linux
